mnefun/_mnefun.py

```diff
@@ -1561,7 +1566,10 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
                                  proj=True, reject=None, flat=None,
                                  preload=True, decim=decim[si])
             new_dict = get_rejection_threshold(temp_epochs)
-            use_reject, use_flat = _restrict_reject_flat(new_dict, p.flat, raw)
+            use_reject = dict()
+            use_reject.update((k, new_dict[k]) for k in p.autoreject_types)
```

Comment:
FYI you can directly do `use_reject = dict((k, new_dict[k]) for k in p.autoreject_types)`